packages/metal-jsx/src/JSXComponent.js

```diff
@@ -34,6 +34,18 @@ class JSXComponent extends Component {
 	}
 
 	/**
+	 * Fires before state batch changes. Provides hook point for modifying
+	 *     state.
+	 * @param {Event} event
+	 * @protected
+	 */
+	handleStateWillChange_(event) {
+		if (event.type !== 'state' && this.willReceiveProps) {
+			this.willReceiveProps(event.changes);
```

Comment:
This is only for JSX, right? We still should document it somehow in the docs, as we do for the other lifecycle methods. I know we are likely updating the site docs with this, but it is fairly easy to miss on code inspection.

Also, same remarks as `willUpdate` about providing a default empty placeholder.